cpp/src/arrow/compute/key_hash_internal.h

```diff
@@ -161,6 +167,12 @@ class ARROW_EXPORT Hashing64 {
   static void HashMultiColumn(const std::vector<KeyColumnArray>& cols, LightContext* ctx,
                               uint64_t* hashes);
 
+  // Clarify the max temp stack usage for HashBatch so the caller could reserve enough
+  // size in advance.
+  static constexpr auto kHashBatchTempStackUsage =
+      (sizeof(uint16_t) + sizeof(uint64_t) + /*extra=*/1) *
```

Comment:
Minor nit: but why does this constant have 2 `sizeof` calls in the calculation but the previous constant has 3 `sizeof` calls?

Reply:
More comments added.